Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp

```diff
@@ -1191,11 +1191,14 @@ void NetworkConnectionToWebProcess::requestStorageAccess(RegistrableDomain&& sub
     completionHandler({ WebCore::StorageAccessWasGranted::Yes, WebCore::StorageAccessPromptWasShown::No, scope, topFrameDomain, subFrameDomain });
 }
 
-void NetworkConnectionToWebProcess::setLoginStatus(RegistrableDomain&& domain, IsLoggedIn loggedInStatus, CompletionHandler<void()>&& completionHandler)
+void NetworkConnectionToWebProcess::setLoginStatus(RegistrableDomain&& domain, IsLoggedIn loggedInStatus, std::optional<WebCore::LoginStatus> lastAuthentication, CompletionHandler<void()>&& completionHandler)
 {
+    if (isLoginStatusAPIRequiresAuthnEnabled() && !lastAuthentication)
```

Comment:
I don't think we should reject here. We can still try to set the login status, but avoid setting the internal "userMediated" flag for this authentication (in other words set authenticationType=unmanaged. 


We might want to reject here if we implement the "userMediated" option to pass to setStatus. If that flag was passed by the RP and contains true, we would reject any requests without a browser mediated ceremony proceeding it (what you are doing here currently.)

Reply:
But we don't have the "userMediated" option yet. Should I consider to implement that option as well? or just changing the authentication type is enough?

Reply:
I think just changing the authentication type for now is enough. We shouldn't error out of the request.